python/ray/tests/resource_isolation/test_resource_isolation_integration.py

```diff
@@ -0,0 +1,184 @@
+import sys
+from pathlib import Path
+
+import pytest
+from click.testing import CliRunner
+
+import ray
+import ray._private.ray_constants as ray_constants
+import ray._private.utils as utils
+import ray.scripts.scripts as scripts
+from ray._private.resource_isolation_config import ResourceIsolationConfig
+
+# These tests is intended to run in CI inside a container.
+# If you want to run this test locally, you will need to create a cgroup that
+# the raylet can manage and delegate to the correct user.
+#
+# Run these commands locally before running the test suite:
+#  sudo mkdir -p /sys/fs/cgroup/resource_isolation_test
+#  sudo chown -R $(whoami):$(whoami) /sys/fs/cgroup/resource_isolation_test/
+#  sudo chmod -R u+rwx /sys/fs/cgroup/resource_isolation_test/
+#  echo $$ | sudo tee /sys/fs/cgroup/resource_isolation_test/cgroup.procs
+#
+# Uncomment the following line.
+# _BASE_CGROUP_PATH = "/sys/fs/cgroup/resource_isolation_test"
+#
+# Comment the following line out.
+_BASE_CGROUP_PATH = "/sys/fs/cgroup"
+
+
+def test_resource_isolation_enabled_creates_cgroup_hierarchy(ray_start_cluster):
+    cluster = ray_start_cluster
+    base_cgroup = _BASE_CGROUP_PATH
+    resource_isolation_config = ResourceIsolationConfig(
+        enable_resource_isolation=True,
+        cgroup_path=base_cgroup,
+        system_reserved_memory=1024**3,
```

Comment:
marking here just as a reminder from our conversation, please tweak this config to include the unit name.